diagram.md

```diff
@@ -2,240 +2,284 @@
-
-Here is a representation of the Block structure. There is an in-depth discussion of it in the specs. I will add some descriptions here when I have time.
-
-One thing that I do want to mention however, to avoid confusion, is the Timecode. The quick eye will notice that there is one Timecode shown per Cluster, and then another within the Block structure itself. The way this works is that the Timecode in the Cluster is relative to the whole file. It is usually the Timecode that the first Block in the Cluster needs to be played at. The Timecode in the Block itself is relative to the Timecode in the Cluster. For example, let's say that the Timecode in the Cluster is set to 10 seconds, and you have a Block in that Cluster that is supposed to be played 12 seconds into the clip; this means that the Timecode in the Block would be set to 2 seconds.
- 
+# Matroska Structure
+
+A Matroska file is composed of one or many `EBML Documents` that use the `Matroska Document Type`. Each `EBML Document` MUST start with an `EBML Header` and then the `Root Element`, which is called `Segment` in Matroska. Matroska defines several `Top Level Elements` which MAY occur within the `Segment`.
+
+As an example, a simple Matroska file consisting of a single `EBML Document` could be represented like this:
+
+- EBML Header
+- Segment
+
+A more complex Matroska file consisting of an `EBML Stream`  (consisting of two `EBML Documents`) could be represented like this:
+
+- EBML Header
+- Segment
+- EBML Header
+- Segment
+
+The following diagram represents a simple Matroska file, comprised of an `EBML Document` with an `EBML Header`, a `Segment Element` (the `Root Element`), and all eight Matroska `Top Level Elements`.
+
+```
++-----------------------+
+| EBML Header           |
++-----------------------|
+| Segment | SeekHead    |
+|         |-------------|
+|         | Info        |
+|         |-------------|
+|         | Tracks      |
+|         |-------------|
+|         | Chapters    |
+|         |-------------|
+|         | Cluster     |
+|         |-------------|
+|         | Cues        |
+|         |-------------|
+|         | Attachments |
+|         |-------------|
+|         | Tags        |
++-----------------------+
+```
+
+## Matroska Top Level Elements
+
+The Matroska `EBML Schema` defines eight `Top Level Elements`: `SeekHead`, `Info`, `Tracks`, `Chapters`, `Cluster`, `Cues`, `Attachments`, and `Tags`.
+
+The `SeekHead Element` (also known as `MetaSeek`) contains an index of where other `Top Level Elements` of the `Segment` are located in order to let the parser know where the other major parts of the file are. This element isn't technicaly REQUIRED, but without a `SeekHead Element` a `Matroska Parser` would have to search the entire file to find all of the other Top Level Elements. This is because Matroska has flexible ordering requirements; for instance, the `Chapters Element` could be stored in the middle of the `Cluster Elements`.
```

Comment:
"…stored in the middle of the `Cluster Elements`" — I'd prefer a more realistic example, one that's actually used in the wild: just change "in the middle of" to "after".